sui_types/src/base_types.rs

```diff
@@ -553,6 +553,66 @@ impl ObjectID {
             .map_err(|_| ObjectIDParseError::TryFromSliceError)
             .map(ObjectID::from)
     }
+
+    /// Incremenent the ObjectID by usize IDs, assuming the ObjectID hex is a number represented as an array of bytes
+    pub fn advance(&self, step: usize) -> Result<ObjectID, anyhow::Error> {
```

Comment:
Random ID generation will lead to huge genesis config files. Ranges allow me compress and partition load gens to only work on specific objects.

So an entry of 1 million object ids looks like this instead of 1M lines of text
```
    {
      "address": "57ce23f7b759259f2831500f637baa59a15a7023",
      "gas_objects": [],
      "gas_object_ranges": [
        {
          "offset": "00000000000000000000000000100000007a1200",
          "count": 1000000,
          "gas_value": 18446744073709551615
        }
      ]
    },
```

Reply:
But you're right. I should try to localize it to the files where its used